modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/validation/SampleDataModelValidator.java

```diff
@@ -144,6 +157,38 @@ private void validateResolver(ConnectableComponentModel model,
     }
   }
 
+  private void validateGenerics(ConnectableComponentModel model, ProblemsReporter problemsReporter,
+                                Class<? extends SampleDataProvider> providerClass) {
+    String providerGenerics = asGenericSignature(getInterfaceGenerics(providerClass, SampleDataProvider.class));
+    String outputGenerics = asGenericSignature(getOutputTypes(model, providerClass.getClassLoader()));
+
+    if (!Objects.equals(providerGenerics, outputGenerics)) {
+      problemsReporter.addError(new Problem(model, format(
+                                                          "SampleDataProvider [%s] was expecting to define '%s' generics signature but '%s' was found instead",
+                                                          providerClass.getName(), outputGenerics, providerGenerics)));
+    }
+  }
+
+  private List<Type> getOutputTypes(ConnectableComponentModel model, ClassLoader classLoader) {
+    return asList(JavaTypeUtils.getType(model.getOutput().getType(), classLoader),
+                  JavaTypeUtils.getType(model.getOutputAttributes().getType(), classLoader));
+  }
+
+  private String asGenericSignature(List<Type> types) {
+    return "<" + types.stream()
+        .map(this::asString)
+        .collect(joining(",")) + ">";
+  }
+
+  private String asString(Type type) {
+    if (type instanceof ParameterizedTypeImpl) {
+      ParameterizedTypeImpl parameterizedType = (ParameterizedTypeImpl) type;
+      return parameterizedType.getRawType().getName() + asGenericSignature(asList(parameterizedType.getActualTypeArguments()));
+    } else {
+      return type != null ? type.getTypeName() : Object.class.getName();
```

Comment:
Maybe when the type is null, and is the generic of the SampleDataProvider, we should add an error that states that the SampleDataProviderGenerics must be explicit.

Reply:
This is for cases in which one of the provider's generic types is a Map or some other parameterized type where we cannot take it upon ourselves to demand proper use of those clases